### PR TITLE
Extracted and named all timeouts and intervals around pings.

### LINKED ISSUE #1212: Fix GSB ping timeout to higher value

### PR DESCRIPTION
**Pull request by maaktweluit**
Increased the timeout to 20s, kept the intervals at 2.5s wait and 5s send

resolves golemfactory/yagna#1212

### CODE REVIEW

File: crates/router/src/lib.rs [hunk 1]
```diff
@@ -20,10 +20,20 @@ use ya_sb_util::PrefixLookupBag;
 mod dispatcher;
 
 lazy_static! {
-    pub static ref GSB_PING_TIMEOUT: u64 = env::var("GSB_PING_TIMEOUT")
-        .unwrap_or("5".into())
+    static ref RAW_GSB_PING_TIMEOUT: u64 = env::var("GSB_PING_TIMEOUT")
+        .unwrap_or(20.to_string())
```
Comment (mfranciszkiewicz) [inline crates/router/src/lib.rs:24]: This change should probably be discussed with @tworec

Comment (maaktweluit) [reply crates/router/src/lib.rs:24]: currently the timeout is 10 ( due to the logic below checking X2 )

without reconnect the small linux machine dies within minutes.

@tworec what do you think of raising this timeout from 10 to 20 seconds?

**Review by mfranciszkiewicz — APPROVED** [review]

File: crates/router/src/lib.rs (hunk 1 above)
Comment (tworec) [reply crates/router/src/lib.rs:24]: I agree
@prekucki suggested similar value as far as I remember our phone call

File: crates/router/src/lib.rs [hunk 2]
```diff
@@ -20,10 +20,20 @@ use ya_sb_util::PrefixLookupBag;
...
         .parse()
         .unwrap();
+    // After GSB_PING_TIMEOUT without reply the connection will disconnect ( default = 20s )
+    pub static ref GSB_PING_TIMEOUT: Duration = Duration::seconds(*RAW_GSB_PING_TIMEOUT as i64);
+
+    // NOTE: Check and try intervals should be lower than ping timeout to avoid race conditions
+    // Try to send a ping to each connection at this interval ( default = 5s )
+    pub static ref GSB_PING_TRY_INT: Duration = *GSB_PING_TIMEOUT / 4;
```
Comment (tworec) [inline crates/router/src/lib.rs]: INT is ambiguous here as a short name of INTERVAL. It rather recalls INTEGER in IT world.

File: crates/router/src/lib.rs [hunk 3]
```diff
@@ -20,10 +20,20 @@ use ya_sb_util::PrefixLookupBag;
 mod dispatcher;
 
 lazy_static! {
-    pub static ref GSB_PING_TIMEOUT: u64 = env::var("GSB_PING_TIMEOUT")
-        .unwrap_or("5".into())
+    static ref RAW_GSB_PING_TIMEOUT: u64 = env::var("GSB_PING_TIMEOUT")
+        .unwrap_or(20.to_string())
         .parse()
         .unwrap();
+    // After GSB_PING_TIMEOUT without reply the connection will disconnect ( default = 20s )
```
Comment (tworec) [inline crates/router/src/lib.rs]: default value within a comment will be outdated (not updated) soonish... pls remove it

**Review by tworec — CHANGES_REQUESTED** [review]

File: crates/router/src/lib.rs (hunk 3 above)
Comment (maaktweluit) [reply crates/router/src/lib.rs]: are we planning to change it again?

Will remove it :)

File: crates/router/src/lib.rs (hunk 2 above)
Comment (maaktweluit) [reply crates/router/src/lib.rs]: i thought the type "Duration" would help distinguish from `INTEGER`, but can change it :)

**Review by tworec — APPROVED** [review]

**Pull request merged**